src/main/java/com/openlattice/chronicle/services/message/MessageService.kt

```diff
@@ -0,0 +1,99 @@
+package com.openlattice.chronicle.services.message
+
+import com.google.common.collect.ArrayListMultimap
+import com.google.common.collect.ImmutableSet
+import com.google.common.collect.ListMultimap
+import com.openlattice.chronicle.constants.EdmConstants
+import com.openlattice.chronicle.data.MessageDetails
+import com.openlattice.chronicle.constants.MessageOutcome
+import com.openlattice.chronicle.services.ApiCacheManager
+import com.openlattice.chronicle.services.edm.EdmCacheManager
+import com.openlattice.chronicle.services.enrollment.EnrollmentManager
+import com.openlattice.chronicle.services.entitysets.EntitySetIdsManager
+import com.openlattice.chronicle.services.twilio.TwilioManager
+import com.openlattice.chronicle.util.ChronicleServerUtil
+import com.openlattice.client.ApiClient
+import com.openlattice.data.DataAssociation
+import com.openlattice.data.DataGraph
+import org.slf4j.LoggerFactory
+import java.time.OffsetDateTime
+import java.util.*
+import java.util.concurrent.ExecutionException
+
+/**
+ * @author toddbergman &lt;todd@openlattice.com&gt;
+ */
+open class MessageService(
+    private val apiCacheManager: ApiCacheManager,
+    private val edmCacheManager: EdmCacheManager,
+    private val enrollmentManager: EnrollmentManager,
+    private val entitySetIdsManager: EntitySetIdsManager,
+    private val twilioManager: TwilioManager
+) : MessageManager {
+
+    companion object {
+        protected val logger = LoggerFactory.getLogger(MessageService::class.java)
+    }
+
+    override fun sendMessages(organizationId: UUID, messageDetailsList: List<MessageDetails>) {
+        val outcomes = twilioManager.sendMessages(messageDetailsList)
+        for(outcome in outcomes) {
+            try {
+                recordMessageSent(organizationId, outcome)
+                logger.info("Recorded notification {} sent to participant {}", outcome.sid, outcome.participantId)
+            } catch (e: ExecutionException) {
+                logger.error("Unable to record notification sent for SID {}", outcome.sid, e)
```

Comment:
What is `sid`? Is that safe to log?

Reply:
it's an id that can be used to retrieve more information about the specific resource. it's a Twilio id. I have never logged into Twilio, but I imagine it's useful in identifying where the message came from.